### PR TITLE
#5337: make Mixtral multidevice+async

### PR DESCRIPTION
**Pull request by sraizada-tt**
replacing device loops with ttnn multi device + async API

### CODE REVIEW

**Comment by mtairum** [comment]
Added a commit for some leftover code on my previous merge (oops).

Started all 3 multichip pipelines. We'll have to check that they are all passing the mixtral tests at least.

- https://github.com/tenstorrent/tt-metal/actions/runs/9098906802
- https://github.com/tenstorrent/tt-metal/actions/runs/9098912081
- https://github.com/tenstorrent/tt-metal/actions/runs/9098913480

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_model.py
Comment (mtairum) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_model.py:49]: Uncomment the pytest parametrizations.

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_rms_norm.py
Comment (mtairum) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_rms_norm.py:59]: Remove print

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_rms_norm.py
Comment (mtairum) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_rms_norm.py:34]: Is this intented or a typo?

**Review by mtairum — CHANGES_REQUESTED** [review]
Please fix this issues before we merge.

**Comment by mtairum** [comment]
New pipelines for the fixes running here:

multi-device unit tests: https://github.com/tenstorrent/tt-metal/actions/runs/9109035704 
multi-device demo: https://github.com/tenstorrent/tt-metal/actions/runs/9109001480
multi-device perf: https://github.com/tenstorrent/tt-metal/actions/runs/9109039877

**Review by yieldthought — APPROVED** [review]

**Comment by mtairum** [comment]
Running updated pipelines:

https://github.com/tenstorrent/tt-metal/actions/runs/9130971955
https://github.com/tenstorrent/tt-metal/actions/runs/9130974546
https://github.com/tenstorrent/tt-metal/actions/runs/9130975564
https://github.com/tenstorrent/tt-metal/actions/runs/9130976749

After they're all green we should confirm the time it takes to run all `test_mixtral_model.py` tests and adjust them to the correct pipelines and/or reduce the number of tests if needed.

**Review by pgkeller — COMMENTED** [review]
Says I'm tagged as code owner, but I don't see any files on the list that I know anything about (looks like all .py?). Let me know if that's blocking at some point and I'll approve

**Comment by tapspatel** [comment]
Added a comment. Rerun pipelines to ensure passing and ill approve.

**Review by cfjchu — APPROVED** [review]
Looks great!

**Comment by mtairum** [comment]
All pipelines passing here:

- [T3K] T3000 model perf tests: https://github.com/tenstorrent/tt-metal/actions/runs/9161206926
- [T3K] T3000 frequent tests: https://github.com/tenstorrent/tt-metal/actions/runs/9161205037
- [T3K] T3000 unit tests: https://github.com/tenstorrent/tt-metal/actions/runs/9161210132
- [T3K] T3000 demo tests: https://github.com/tenstorrent/tt-metal/actions/runs/9172485831

I'll squash some commits and merge.

**Review by mtairum — APPROVED** [review]

**Review by tapspatel — APPROVED** [review]

**Pull request merged**
